filebeat/harvester/processor/line.go

```diff
@@ -0,0 +1,90 @@
+package processor
+
+import (
+	"io"
+	"time"
+
+	"github.com/elastic/beats/filebeat/harvester/encoding"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+// LineSource produces lines by reading lines from an io.Reader
+// through a decoder converting the reader it's encoding to utf-8.
+type LineSource struct {
+	reader *encoding.LineReader
+}
```

Comment:
`LineSource` is basically an adapter for converting an `io.Reader` into a `LineProcessor` producing lines. Alternative names `LinesAdapter` or `LineProducer`. Perfectly valid to move LineSource into separate go file and only define interfaces in `line.go`.